### PR TITLE
Issue 187#: Long duration test for data service

### PR DESCRIPTION
**Pull request by yamingk**
1. add long duration test for data service with random read, write and free. 
2. fixed one issue in MultiBlkId iterator
3. add/update api comments for data service

Testing:
======
a few hundred iterations run for the mixed io load (num_io=300) test case and passed. --- which included in conan build test.
Long duration run for a few hours and still running.

### CODE REVIEW

File: src/lib/blkalloc/blk.cpp [hunk 1]
```diff
@@ -128,6 +128,7 @@ std::string MultiBlkId::to_string() const {
     auto it = iterate();
     while (auto const b = it.next()) {
         str += b->to_string();
+        str += " ";
```
Comment (hkadayam) [inline src/lib/blkalloc/blk.cpp:131]: nit: could fold this into one line by 
```str += b->to_string() + " "```

File: src/tests/test_data_service.cpp [hunk 2]
```diff
@@ -56,13 +60,15 @@ constexpr uint64_t Mi{Ki * Ki};
...
+    uint64_t num_io{1000};
+    uint64_t run_time{200};        // in seconds
+    uint32_t min_io_size{4 * Ki};  // blk_size aligned;
+    uint32_t max_io_size{32 * Mi}; // blk_size aligned;
 };
 
 static Param gp;
 
-ENUM(DataSvcOp, uint8_t, WRITE, READ, FREE_BLK, COMMIT_BLK, RESERVE_STREAM, ALLOC_STREAM, FREE_STREAM)
+ENUM(DataSvcOp_t, uint8_t, async_alloc_write = 1, async_read = 2, async_free = 3, max_op = 4);
```
Comment (hkadayam) [inline src/tests/test_data_service.cpp:71]: If you want to allocate the value for it we should use `VENUM(..`. I don't see much benefit in assigning the value here.

Comment (yamingk) [reply src/tests/test_data_service.cpp:71]: acked

File: src/lib/blkalloc/blk.cpp (hunk 1 above)
Comment (yamingk) [reply src/lib/blkalloc/blk.cpp:131]: done

**Review by hkadayam — APPROVED** [review]

**Pull request merged**